app/addons/fauxton/components.react.jsx

```diff
@@ -334,13 +354,246 @@ function (app, FauxtonAPI, React, ZeroClipboard) {
+            </footer>
+          </div>
+
+          <div className={maskClasses} onClick={Actions.hideNotificationCenter}></div>
+        </div>
+      );
+    }
+  });
+
+  var NotificationRow = React.createClass({
+    propTypes: {
+      item: React.PropTypes.object.isRequired,
+      filter: React.PropTypes.string.isRequired,
+      transitionSpeed: React.PropTypes.number
+    },
+
+    getDefaultProps: function () {
+      return {
+        transitionSpeed: 300
+      };
+    },
+
+    clearNotification: function () {
+      var notificationId = this.props.item.notificationId;
+      this.hide(function () {
+        Actions.clearSingleNotification(notificationId);
+      });
+    },
+
+    componentDidMount: function () {
+      this.setState({
+        elementHeight: this.getHeight()
+      });
+    },
+
+    componentDidUpdate: function (prevProps) {
+      // in order for the nice slide effects to work we need a concrete element height to slide to and from.
+      // $.outerHeight() only works reliably on visible elements, hence this additional setState here
+      if (!prevProps.isVisible && this.props.isVisible) {
+        this.setState({
+          elementHeight: this.getHeight()
+        });
+      }
+
+      var show = true;
+      if (this.props.filter !== 'all') {
+        show = this.props.item.type === this.props.filter;
+      }
+      if (show) {
+        console.log(this.state.elementHeight);
+        $(this.getDOMNode()).velocity({ opacity: 1, height: this.state.elementHeight }, this.props.transitionSpeed);
+      } else {
+        this.hide();
+      }
+    },
+
+    getHeight: function () {
+      return $(this.getDOMNode()).outerHeight(true);
+    },
+
+    hide: function (onHidden) {
+      $(this.getDOMNode()).velocity({ opacity: 0, height: 0 }, this.props.transitionSpeed, function () {
+        if (onHidden) {
+          onHidden();
+        }
+      });
+    },
+
+    render: function () {
+      var iconMap = {
+        success: 'fonticon-ok-circled',
+        error: 'fonticon-attention-circled',
+        info: 'fonticon-info-circled'
+      };
+
+      var timeElapsed = this.props.item.time.fromNow();
+
+      // we can safely do this because the store ensures all notifications are of known types
+      var rowIconClasses = 'fonticon ' + iconMap[this.props.item.type];
+      var classes = 'flex-layout flex-row';
+
+      // for testing purposes
+      var visible = (this.props.filter === 'all' || this.props.filter === this.props.item.type) ? 'true' : 'false';
+
+      // N.B. wrapper <div> needed to ensure smooth hide/show transitions
+      return (
+        <li data-visible={visible}>
+          <div className={classes}>
+            <span className={rowIconClasses}></span>
+            <div className="flex-body">
+              <p dangerouslySetInnerHTML={{__html: this.props.item.msg}}></p>
```

Comment:
Drat, excellent point.

I liked passing in the HTML here because some messages contained links that would continue to work from within the notification centre panel. But on further thought, it could backfire... not just with the XSS - but with links that were relative, or whatever. Potentially there'd be broken links appearing in the notification centre.

I'll change this to strip out HTML + just pass the msg itself.